FirebaseDatabase/Tests/Unit/Swift/DatabaseAPITests.swift

```diff
@@ -0,0 +1,598 @@
+          } catch {
+            // ...
+          }
+        }
+      }
+    #endif // swift(>=5.5)
+    // onDisconnectUpdateChildValues
+    databaseReference.onDisconnectUpdateChildValues([AnyHashable: Any]())
+    databaseReference
+      .onDisconnectUpdateChildValues([AnyHashable: Any]()) { (
+        optionalError: Error?,
+        databaseReference: DatabaseReference
+      ) in
+      // ...
+      }
+    #if swift(>=5.5)
+      if #available(iOS 15.0, macOS 12.0, macCatalyst 15.0, tvOS 15.0, *) {
+        // async/await is a Swift 5.5+ feature available on iOS 15+
+        async {
+          do {
+            _ = try await databaseReference
+              .onDisconnectUpdateChildValues([AnyHashable: Any]()) as DatabaseReference
+          } catch {
+            // ...
+          }
+        }
+      }
+    #endif // swift(>=5.5)
+    // cancelDisconnectOperations
+    databaseReference.cancelDisconnectOperations()
+    databaseReference
+      .cancelDisconnectOperations { (optionalError: Error?, databaseReference: DatabaseReference) in
+        // ...
+      }
+    #if swift(>=5.5)
+      if #available(iOS 15.0, macOS 12.0, macCatalyst 15.0, tvOS 15.0, *) {
+        // async/await is a Swift 5.5+ feature available on iOS 15+
+        async {
+          do {
+            _ = try await databaseReference.cancelDisconnectOperations() as DatabaseReference
+          } catch {
+            // ...
+          }
+        }
+      }
+    #endif // swift(>=5.5)
+    // runTransactionBlock
+    databaseReference.runTransactionBlock { (mutableData: MutableData) in
+      TransactionResult()
+    }
+    databaseReference.runTransactionBlock { (mutableData: MutableData) in
+      TransactionResult()
+    } andCompletionBlock: { (optionalError: Error?, bool: Bool, optinalDataSnapshot: DataSnapshot?) in
+      // ...
+    }
+    #if swift(>=5.5)
+      if #available(iOS 15.0, macOS 12.0, macCatalyst 15.0, tvOS 15.0, *) {
+        // async/await is a Swift 5.5+ feature available on iOS 15+
+        async {
+          do {
+            _ = try await databaseReference.runTransactionBlock { (mutableData: MutableData) in
+              TransactionResult()
+            } as (Bool, DataSnapshot)
+          } catch {
+            // ...
+          }
+        }
+      }
+    #endif // swift(>=5.5)
+    databaseReference.runTransactionBlock({ (mutableData: MutableData) in
+                                            TransactionResult()
+                                          },
+                                          andCompletionBlock: { (
+                                            optionalError: Error?,
+                                            bool: Bool,
+                                            optionalDataSnapshot: DataSnapshot?
+                                          ) in
+                                          // ...
+                                          }, withLocalEvents: true as Bool)
```

Comment:
Curious, are you able to use this API using the Swift's multiple trailing closure syntax? This [doc](https://github.com/apple/swift-evolution/blob/main/proposals/0279-multiple-trailing-closures.md) shows what they look like.

It'd probably look like this:
```swift
databaseReference.runTransactionBlock { (mutableData: MutableData) in 
  TransactionResult()
}, { (optionalError: Error?, bool: Bool, optionalDataSnapshot: DataSnapshot?) in
  // ...
}, withLocalEvents: true as Bool)
```

Reply:
I don't think I can because the last named parameter is not a closure but rather a `Bool`, though I'll investigate further...

Reply:
Ah, good point. Ok, cool, no worries. I dont know enough about the API but this could be an example where the API could be improved to be like: `func runTransactionBlock(localEvents: Bool, transactionHandler: blah, completion: blah)`